test/Microsoft.IdentityModel.Validators.Tests/AadTokenValidationParametersExtensionTests.cs

```diff
@@ -19,11 +19,187 @@
+                    ValidateAudience = false,
+                    ValidateLifetime = false
+                },
+                ExpectedValidationResult = true,
+            });
+
+            issClaim = new Claim(ValidatorConstants.ClaimNameIss, "different issue");
+            jwk = KeyingMaterial.CreateJsonWebKeyEC(JsonWebKeyECTypes.P256, Default.SymmetricSigningKey.KeyId, KeyingMaterial.P256_D, KeyingMaterial.P256_X, KeyingMaterial.P256_Y);
+            jwk.AdditionalData.Add(OpenIdProviderMetadataNames.Issuer, ValidatorConstants.AadIssuer);
+            signingKeysConfig = SetupOpenIdConnectConfigurationMock(jwk);
+            signingKeysConfig.Issuer = ValidatorConstants.AadIssuer;
+            theoryData.Add(new EnableAadSigningKeyValidationTheoryData
+            {
+                TestId = "IssuerSigningKeyValidatorUsingConfiguration_IssuerNotMatched_ValidationFailed",
+                Token = Default.Jwt(Default.SecurityTokenDescriptor(KeyingMaterial.DefaultSymmetricSigningCreds_256_Sha2, new List<Claim> { issClaim, tidClaim })),
+                TokenValidationParameters = new TokenValidationParameters()
+                {
+                    ConfigurationManager = new StaticConfigurationManager<OpenIdConnectConfiguration>(signingKeysConfig),
+                    IssuerSigningKey = Default.SymmetricSigningKey,
+                    ValidateIssuerSigningKey = true,
+                    ValidateAudience = false,
+                    ValidateLifetime = false
+                },
+                ExpectedValidationResult = false,
+            });
+
+            OpenIdConnectConfiguration SetupOpenIdConnectConfigurationMock(JsonWebKey jsonWebKey = null)
+            {
+                OpenIdConnectConfiguration config = new OpenIdConnectConfiguration() { TokenEndpoint = Default.Issuer + "oauth/token", Issuer = Default.Issuer };
+                config.SigningKeys.Add(KeyingMaterial.DefaultX509Key_2048);
+                config.JsonWebKeySet = new JsonWebKeySet();
+                if (jsonWebKey != null)
+                {
+                    config.JsonWebKeySet.Keys.Add(jsonWebKey);
+                }
+                return config;
+            };
+
+            TokenValidationParameters SetupTokenValidationParametersMock(OpenIdConnectConfiguration configuration) =>
+            new TokenValidationParameters()
+            {
+                ConfigurationManager = new StaticConfigurationManager<OpenIdConnectConfiguration>(configuration),
+                ValidateIssuerSigningKey = true,
+                ValidateAudience = false,
+                ValidateLifetime = false
+            };
+
+            return theoryData;
+        }
+
         [Theory, MemberData(nameof(EnableAadSigningKeyIssuerValidationTestCases))]
-        public async Task EnableAadSigningKeyIssuerValidationTests(AadSigningKeyIssuerTheoryData theoryData)
+        [Obsolete("The 'EnableAadSigningKeyIssuerValidation' extension method is obsolete.")]
```

Comment:
Can we suppress the obsolete warning here (CS0618) for consistence?

Reply:
Fixed